src/core/lib/transport/metadata_batch.h

```diff
@@ -184,6 +187,49 @@ struct GrpcTagsBinMetadata : public SimpleSliceBasedMetadata {
   static absl::string_view key() { return "grpc-tags-bin"; }
 };
 
+template <typename Int>
+struct SimpleIntBasedMetadataBase {
```

Comment:
Is there a reason that this is separate from `SimpleIntBasedMetadata`?  I don't see anywhere else that it's used, but maybe you have something in mind for a subsequent PR?

Reply:
Added a comment explaining - basically if we do not do this, we end up duplicating some code - allowing the compiler to "see" that these are really really the same function helped some bloat.